### PR TITLE
Reword “Managing Compute Resources for Containers”

### PR DESCRIPTION
**Pull request by sftim**
Reword [Managing Compute Resources for Containers](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/):
- change the title to “Managing Resources for Containers”, because there are other resource that aren't compute
  - maybe “Requests and limits” would be a better title. What do people think?
- redo the page introduction
- move troubleshooting to the end of the page
- rewrite the section about _local ephemeral storage_
- tweak the What's next? section

Mark monitoring / management of local ephemeral storage as beta since v1.10 (rather than just the current version)

I really didn't understand the existing text for local ephemeral storage, so I've tried to reword it to clarify how I _think_ it works. I've aimed to state things clearly so that if I have made an incorrect assumption, that shows up and we can fix it.

I also moved the troubleshooting section to the foot of the concept page. Really I think that might work better as a Task, but for now moving it to the end is at least a small improvement.

This really does want tech review from someone familiar with the local ephemeral storage feature, as well as with this overall area of Kubernetes.

/sig storage
/sig node

### CODE REVIEW

**Comment by bradtopol** [comment]
This is pretty large. Would like a few pairs of eyes on this. Assigning @jimangel and myself
/assign @jimangel

**Comment by sftim** [comment]
If it helps, you could review by commit and sign off on the ones that look fine, possibly leaving 1 or 2 commits that still need more eyes on them.

**Comment by sftim** [comment]
I _think_ [9bb489..d770089](https://github.com/kubernetes/website/pull/18425/files/5010f7dd8fe5bb2b25af5cadcc3941c4e9c57b98..d7700896e95b8e93287acdad56926b8b26a7fa5d) shouldn't be controversial; it's [5010f7d](https://github.com/kubernetes/website/pull/18425/commits/5010f7dd8fe5bb2b25af5cadcc3941c4e9c57b98) that I was particular unclear on.

File: content/en/docs/concepts/configuration/manage-compute-resources-container.md [hunk 1]
```diff
@@ -360,7 +369,7 @@ Mi, Ki. For example, the following represent roughly the same value:
 128974848, 129e6, 129M, 123Mi
 ```
 
-For example, the following Pod has two Containers. Each Container has a request of 2GiB of local ephemeral storage. Each Container has a limit of 4GiB of local ephemeral storage. Therefore, the Pod has a request of 4GiB of local ephemeral storage, and a limit of 8GiB of storage.
+In the following example, the Pod has two Containers. Each Container has a request of 2GiB of local ephemeral storage. Each Container has a limit of 4GiB of local ephemeral storage. Therefore, the Pod has a request of 4GiB of local ephemeral storage, and a limit of 8GiB of storage.
```
Comment (zacharysarah) [inline content/en/docs/concepts/configuration/manage-compute-resources-container.md]: Nice change. 👍

File: content/en/docs/concepts/configuration/manage-compute-resources-container.md [hunk 2]
```diff
@@ -397,71 +406,94 @@ The scheduler ensures that the sum of the resource requests of the scheduled Con
...
 
-XFS filesystems require no special action when building; they are
-automatically built with project quotas enabled.
+{{< feature-state for_k8s_version="v1.15" state="alpha" >}}
 
-Ext4fs filesystems must be built with quotas enabled, then they must
-be enabled in the filesystem:
+Project quotas are an operating-system level feature for managing
+storage use on filesystems. With Kubernetes, you can enable project
+quotas for monitoring storage use. You need to make sure that the filesystem
```
Comment (zacharysarah) [inline content/en/docs/concepts/configuration/manage-compute-resources-container.md]: "Make sure that the filesystem..."

File: content/en/docs/concepts/configuration/manage-compute-resources-container.md [hunk 3]
```diff
@@ -397,71 +406,94 @@ The scheduler ensures that the sum of the resource requests of the scheduled Con
...
+backing the emptyDir volumes, on the node, provides project quota support.
+For example, XFS and ext4fs can offer project quotas.
 
-```
-% sudo mkfs.ext4 other_ext4fs_args... -E quotatype=prjquota /dev/block_device
-% sudo tune2fs -O project -Q prjquota /dev/block_device
+{{< note >}}
+Project quotas let you monitor storage use; they do not enforce limits
```
Comment (zacharysarah) [inline content/en/docs/concepts/configuration/manage-compute-resources-container.md]: Nit: period after limits

File: content/en/docs/concepts/configuration/manage-compute-resources-container.md [hunk 4]
```diff
@@ -397,71 +406,94 @@ The scheduler ensures that the sum of the resource requests of the scheduled Con
...
-```
+Kubernetes uses project IDs starting from `1048576`.  The IDs in use are
+registered in `/etc/projects` and `/etc/projid`.  If project IDs in
+this range are used for other purposes on the system, those project
+IDs must be registered in `/etc/projects` and `/etc/projid` so that
+Kubernetes does not use them.
 
-To mount the filesystem, both ext4fs and XFS require the `prjquota`
-option set in `/etc/fstab`:
+Quotas are faster and more accurate than directory scanning.  When a
```
Comment (zacharysarah) [inline content/en/docs/concepts/configuration/manage-compute-resources-container.md]: Nit: no double spaces after periods

File: content/en/docs/concepts/configuration/manage-compute-resources-container.md [hunk 5]
```diff
@@ -335,18 +335,27 @@ LastState: map[terminated:map[exitCode:137 reason:OOM Killed startedAt:2015-07-0
...
+<!-- feature gate LocalStorageCapacityIsolation -->
+{{< feature-state for_k8s_version="v1.10" state="beta" >}}
 
-This partition is “ephemeral” and applications cannot expect any performance SLAs (Disk IOPS for example) from this partition. Local ephemeral storage management only applies for the root partition; the optional partition for image layer and writable layer is out of scope.
+Nodes have local ephemeral storage, backed either by RAM or by locally-attached writeable devices. The kubelet uses local ephemeral storage to provide emptyDir {{< glossary_tooltip term_id="volume" text="volumes" >}} to Pods. The kubelet also uses this kind of storage to hold [node-level container logs](/docs/concepts/cluster-administration/logging/#logging-at-the-node-level), container images, and the writable layers of running containers.
+
+“Ephemeral” means that there is no long-term guarantee about durability. If a Node fails, the data in its ephemeral storage can be lost. Your applications cannot expect any performance SLAs (disk IOPS for example) from local ephemeral storage.
```
Comment (zacharysarah) [inline content/en/docs/concepts/configuration/manage-compute-resources-container.md]: Because this warns about potential data loss with no SLA, I'd like to see the second and third sentences become a separate paragraph enclosed in a `{{< warning >}}`.

**Comment by zacharysarah** [comment]
@sftim 👋 I'd love to see this PR get to merge. Please feel free to `/reopen` this when you're ready to continue!

/close

**Comment by sftim** [comment]
/reopen
(don't think this works for PRs though)

**Comment by sftim** [comment]
Ah, ordering problem.
